File: Backend/Employee/Employee/Service/EmployeeService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Employee.Service
{
    public class EmployeeService : IEmployeeService
    {
        public IEnumerable<Employee> GetEmployee()
        {
            return EmployeeModel.EmployeeList;
        }


        public void AddEmployee(Employee employee)
        {
            var serialNo = EmployeeModel.EmployeeList.Count + 1;
            employee.Sno = serialNo;
            EmployeeModel.EmployeeList.Add(employee);
        }


        public void EditEmployee(Employee employee)
        {
            var empObj = EmployeeModel.EmployeeList.FirstOrDefault(e => e.Sno == employee.Sno);
            empObj.Sno = employee.Sno;
            empObj.FirstName = employee.FirstName;
            empObj.LastName = employee.LastName;
            empObj.Team = employee.Team;
            empObj.Area = employee.Area;
            empObj.Title = employee.Title;
            empObj.DateOfJoining = employee.DateOfJoining;
        }


        public void DeleteEmployee(int id)
        {
            var empObj = EmployeeModel.EmployeeList.FirstOrDefault(e => e.Sno == id);
            EmployeeModel.EmployeeList.Remove(empObj);
        }
    }
}

File: Frontend/src/app/components/Employee/employeeService.js
'use strict';
class Employees {
   
	constructor($http,$q) {
		this.$http=$http;
		this.employeeList=[];
		this.$q=$q;
	}
	
	getListOfEmployee() {
		var deferred = this.$q.defer();
		this.$http.get('http://localhost:2771/api/employee/get').
				then((response)=> {
					this.employeeList=response.data;
					deferred.resolve(this.employeeList);
				}, function(response) {
					deferred.reject();
				});
		return deferred.promise;
	}
	
	addEmployee(employee) {
		  var deferred = this.$q.defer();
		  var request = this.$http.post('http://localhost:2771/api/employee/add', {
				FirstName: employee.FirstName,
				LastName: employee.LastName,
				Team:employee.Team,
				Area: employee.Area,
				Title: employee.Title,
				DateOfJoining:employee.DateOfJoining
        });
		request.success((employee)=>{
			deferred.resolve();
		});
		return deferred.promise;
	}
	
	deleteEmployee(employee) {
		var deferred = this.$q.defer();
		this.$http.get(employee.Actions[2].Url,{data :employee.Sno}).then(()=>{
			var index = this.employeeList.indexOf(employee);
			this.employeeList.splice(index, 1);
			deferred.resolve();
	 });
		return deferred.promise;
	}

	editEmployee(employee) {
	     var deferred = this.$q.defer();
		 var request = this.$http.post('http://localhost:2771/api/Employee/edit', {
                Sno:employee.Sno,
				FirstName: employee.FirstName,
				LastName: employee.LastName,
				Team:employee.Team,
				Area: employee.Area,
				Title: employee.Title,
				DateOfJoining:employee.DateOfJoining
        });
		request.success(()=>{
			deferred.resolve();
		});
		return deferred.promise;
	}

}
Employees.$inject = ['$http','$q'];

angular
	.module('myAngularApp')
	.service('EmployeeService', Employees);
